### PR TITLE
command rework

### PR DESCRIPTION
**Pull request by joerivanarkel**

### CODE REVIEW

**Comment by joerivanarkel** [comment]
**Summary**:
- Refined and Refactored commands
- Added a dependency injection container, from Application to the Parser and Lexer
- Moved constant variable parameters to `variables.json` and added new `EnvironmentVariables` class to get these variables

File: Common/EnvironmentVariables.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            {
+                var directory = Directory.GetCurrentDirectory();
+                var secretConfig = new ConfigurationBuilder()
+                    .AddJsonFile($"{directory}/../Shell/variables.json", false, true)
+                    .Build();
+                return secretConfig[variablename];
+            }
+            catch (System.Exception)
+            {
+                throw new NotImplementedException($"{variablename} was not found in variables.json");
```
Comment (teusvanarkel) [inline Common/EnvironmentVariables.cs:19]: That's not the right exception....create a custom exception for variable not found

File: Common/Logger.cs [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
+    private string version { get; set; } = EnvironmentVariables.GetVariable("version");
     private string location { get; set; }
 
-    public Logger(string Version, string Name = "placeholder")
+    public Logger(string Name = "placeholder")
     {
-        version = Version;
-
-
         var dateTime = DateTime.Now;
```
Comment (teusvanarkel) [inline Common/Logger.cs:16]: Why?

File: Common/ObjectSerializer.cs [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
     public string Location { get; set; }
     public ILogger Logger { get; set; }
-    public ObjectSerializer(string location, ILogger logger)
+    public ObjectSerializer(ILogger logger)
     {
-        Location = $"{location}/cache";
+        Location = EnvironmentVariables.GetVariable("location") + "/cache";
         bool exists = System.IO.Directory.Exists(Location);
         if (!exists)
             System.IO.Directory.CreateDirectory(Location);
```
Comment (teusvanarkel) [inline Common/ObjectSerializer.cs:17]: Maybe create a fileHandler or fileManager to do the file handling logic and get it out of the serializer class ( seperation of concerns )

File: Shell/Program.cs [hunk 4]
```diff
@@ -1,70 +1,36 @@
-﻿// See https://aka.ms/new-console-template for more information
-using Shell.Commands;
+﻿using System.Windows.Input;
 using Common;
-using Common.Enum;
+using Common.Interfaces;
```
Comment (teusvanarkel) [inline Shell/Program.cs:3]: Move to implicit usings?

File: Shell/Program.cs [hunk 5]
```diff
@@ -1,70 +1,36 @@
...
+builder.AddTransient<ICommandFactory, CommandFactory>();
 
-        foreach (var item in args)
-        {
-            if (item != null)
-            {
-                argumentsList.Add(item);
-            }
-        }
+builder.AddTransient<ILexer, Lexer.Lexer>();
```
Comment (teusvanarkel) [inline Shell/Program.cs:29]: Are there no places left where the Lexer is created?

**Review by teusvanarkel — COMMENTED** [review]
Super good progressing...some comments!.

File: Common/ObjectSerializer.cs (hunk 3 above)
Comment (joerivanarkel) [reply Common/ObjectSerializer.cs:17]: #65

File: Shell/Program.cs (hunk 4 above)
Comment (joerivanarkel) [reply Shell/Program.cs:3]: #64

**Pull request merged**